src/reflect/scala/reflect/internal/JDK9Reflectors.java

```diff
@@ -0,0 +1,54 @@
+package scala.reflect.internal;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.jar.JarFile;
+
+public final class JDK9Reflectors {
+    private static final MethodHandle RUNTIME_VERSION_PARSE;
+    private static final MethodHandle NEW_JAR_FILE;
+
+    static {
+        RUNTIME_VERSION_PARSE = lookupRuntimeVersionParse();
+        NEW_JAR_FILE = lookupNewJarFile();
+    }
+
+    public static /*java.lang.Runtime.Version*/ Object lookupRuntimeVersionParse(String string) {
```

Comment:
> Also this method could be made private and invoked directly from `newJarFile`.

I'd prefer to keep these methods signature identical to the methods they reflect, rather than mixing concerns of the reflection with facades.